libbeat/common/string.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+type NetString []byte
+
+// implement encoding.TextMarshaller interface to treat []byte as raw string
+// by other encoders/serializers (e.g. JSON)
+
+func (n NetString) MarshalText() ([]byte, error) {
+	return n, nil
+}
+
+func (n NetString) UnsafeString() string {
+	bh := (*reflect.SliceHeader)(unsafe.Pointer(&n))
+	sh := reflect.StringHeader{Data: bh.Data, Len: bh.Len}
+	return *(*string)(unsafe.Pointer(&sh))
```

Comment:
Go performs a copy when converting from `[]byte` to `string` to guarantee that the contents of the `string` can not be changed by mutating the `[]byte` since the contract in Go is that strings are immutable.

This seems like a good optimization to use when you *know* for sure that the returned string is not going to be retained (where immutability guarantees might be required). The method should get updated to document when and how to safely use it.